src/app/task/task.service.ts

```diff
@@ -0,0 +1,65 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Task } from 'src/entities/task.entity';
+import { Repository } from 'typeorm';
+import { TaskDto } from './dto/task.dto';
+import { User } from 'src/entities/user.entity';
+
+@Injectable()
+export class TaskService {
+  constructor(
+    @InjectRepository(Task)
+    private taskRepository: Repository<Task>,
+  ) {}
+
+  async getDefaultTaskList(
+    workspaceId: number,
+    user: User,
+    isForCurrentUserOnly = false,
+  ): Promise<{ tasks: Task[]; count: number }> {
+    const qb = this.taskRepository
+      .createQueryBuilder('tasks')
+      .leftJoinAndSelect('tasks.user', 'user')
+      .select('*')
+      .where('tasks.summary = :summary', { summary: null })
+      .andWhere('tasks.workspace = :workspaceId', { workspaceId });
+    if (isForCurrentUserOnly) {
+      qb.andWhere('tasks.user = :userId', { user });
+    }
+
+    const [tasks, count] = await qb.getManyAndCount();
+    return { tasks, count };
+  }
+
+  async createTask(user: User, dto: TaskDto): Promise<Task> {
+    return await this.taskRepository.save({
+      user,
+      name: dto.name,
+      status: dto.status,
+      workspaceId: dto.workspaceId,
```

Comment:
workspace instead of workspaceId - and it would make sense to do the same validation as above